### PR TITLE
remove breadcrumb library

### LINKED ISSUE #3168: Maintenance: remove breadcrumb library
### Description
While exploring a bug in the breadcrumbs: https://github.com/sul-dlss/exhibits/issues/2585 a look at the [breadcrumbs library](https://github.com/weppos/breadcrumbs_on_rails) shows it hasn't been updated in 2 years.

An investigation shows that moving breadcrumbs into spotlight code requires minimal work.

### PR DESCRIPTION
**Pull request by dnoneill**
closes #3168

### CODE REVIEW

File: app/views/shared/_breadcrumbs.html.erb [hunk 1]
```diff
@@ -1,7 +1,25 @@
-<div class="container breadcrumbs-container" role="navigation" aria-label="<%= t('spotlight.breadcrumb.label') %>">
-  <div class="row">
-    <div class="col-md-12">
-      <%= render_breadcrumbs builder: Spotlight::BootstrapBreadcrumbsBuilder %>
+<% if defined?(breadcrumbs) && breadcrumbs.present? %>
```
Comment (dnoneill) [inline app/views/shared/_breadcrumbs.html.erb:1]: This got added because spec/features/confirm_email_spec.rb:8 keeps failing without it.

**Comment by hackartisan** [comment]
Can this PR please reference an issue with information about what / why?

**Comment by dnoneill** [comment]
@hackartisan Linked.

File: app/views/shared/_breadcrumbs.html.erb (hunk 1 above)
Comment (dnoneill) [reply app/views/shared/_breadcrumbs.html.erb]: Following this ```

I think if you define these in Spotlight::Controller instead and add breadcrumbs to the list of helpers in the included do block also in Spotlight::Controller you could remove the separate helper_method :breadcrumbs entries in CatalogController and ApplicationController.
```

Fixed the problem. I already had a component already made so I just added it anyway.

File: app/components/breadcrumbs_component.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+# Component to render breadcrumbs
+class BreadcrumbsComponent < ViewComponent::Base
```
Comment (corylown) [inline app/components/breadcrumbs_component.rb:4]: This and its template should probably go in `app/components/spotlight` and be wrapped in `module Spotlight` like all the other spotlight components.

File: app/controllers/spotlight/catalog_controller.rb [hunk 3]
```diff
@@ -8,6 +8,8 @@ module Spotlight
   # rubocop:disable Metrics/ClassLength
   class CatalogController < ::CatalogController
     include Spotlight::Concerns::ApplicationController
+    helper_method :breadcrumbs
+
```
Comment (corylown) [inline app/controllers/spotlight/catalog_controller.rb:12]: I don't think you need this anymore now that `helper_method :breadcrumbs` is set in `Spotlight::Controller`
```suggestion
```

File: app/controllers/concerns/spotlight/controller.rb [hunk 4]
```diff
@@ -13,6 +13,7 @@ module Controller
     included do
       helper_method :current_site, :current_exhibit, :current_masthead, :exhibit_masthead?, :resource_masthead?
       before_action :set_exhibit_locale_scope, :set_locale
+      helper_method :breadcrumbs
```
Comment (corylown) [inline app/controllers/concerns/spotlight/controller.rb:16]:
```suggestion
      helper_method :breadcrumbs, :current_site, :current_exhibit, :current_masthead, :exhibit_masthead?, :resource_masthead?
      before_action :set_exhibit_locale_scope, :set_locale
```

File: app/components/breadcrumbs_component.rb (hunk 2 above)
Comment (corylown) [reply app/components/breadcrumbs_component.rb:4]: Add a test for the component?

**Review by corylown — CHANGES_REQUESTED** [review]

**Review by corylown — APPROVED** [review]

**Pull request merged**